src/runtime/async_std.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{Error, Response};
+
+use std::{sync::Arc, time::Duration};
+
+use crate::mdns::mDNSSender;
+use async_std::future::TimeoutError;
+use futures_core::{Future, Stream};
+use futures_util::{future::ready, stream::select, StreamExt};
+
+pub use async_std::net::UdpSocket as AsyncUdpSocket;
+
+pub fn discovery_listen(
```

Comment:
I think I'd like to clean this up, right now there's way too much duped code between the tokio and async-std implementations so if you wanna hold off on this PR that's fine, it shouldn't take long to consolidate